rfcs/xxxx-v5-content-api.md

```diff
@@ -0,0 +1,217 @@
+- Start Date: 2023-09-05
+- RFC PR: (leave this empty)
+
+# Summary
+
+V5 is in preparation and will require making changes to the Content APIs (REST & GraphQL).
+
+# Motivation
+
+V5 Content Engine requires some breaking changes to the API, It is also the opportunity to simplify the API Response format following user feedbacks.
+
+> The main goal of this RFC is to propose a simplified and v5 compatible format while offering a smoother migration path with a legacy format.
+
+# Detailed design
+
+### Endpoints
+
+Based on the [Database changes planned](https://github.com/strapi/rfcs/pull/52),we will now mainly work with the `documentId` within the API.
```

Comment:
I am curious about how users can make modifications to a particular entry using the API, given that it accepts documentId as input. For example, let's say I want to delete my entry with entryId : how would it be done with the new API ? (Similar questions can be raised for the other endpoints.)

One potential approach to address this issue (if it's considered as one) could be to introduce a new endpoint. Either /api/:contentType/document/:documentId to keep the old API working as is, or add /api/:contentType/entry/:entryId to at least have an endpoint similar to the old behavior.

From my current understanding, multiple entries can refer to the same documentId, and a "document" instance doesn't truly exist by itself (e.g., there isn't a structure like { documentId: '123', entries: [...] }).

If I misunderstood any aspect (I don't have deep expertise in the specifics of your codebase), feel free to ignore my message, though your clarifications will be appreciated. :)